packages/react-dom/src/events/ChangeEventPlugin.js

```diff
@@ -228,17 +229,15 @@ function handleControlledInputBlur(inst, node) {
   }
 
   // Fiber and ReactDOM keep wrapper state in separate places
+  // TODO: Is this still necessary now that Stack is gone?
   let state = inst._wrapperState || node._wrapperState;
 
   if (!state || !state.controlled || node.type !== 'number') {
     return;
   }
 
   // If controlled, assign the value attribute to the current value on blur
-  let value = '' + node.value;
-  if (node.getAttribute('value') !== value) {
-    node.setAttribute('value', value);
-  }
+  synchronizeDefaultValue(node, node.type, node.value);
```

Comment:
@gaearon new here: I forgot about the number behavior on blur when we change inputs. This is so that chrome doesn't validate number inputs when we assign the default value. The DOM Operations are now consistent.